### PR TITLE
Throw exception when disable read thread pool and enable late materialization

### LINKED ISSUE #7486: Throw exception when not enable read thread pool
## Enhancement
Late materialization is incompatible with disabling the read thread pool, so throw an exception when not enabling the read thread pool.

### PR DESCRIPTION
**Pull request by Lloyd-Pottiger**
### What problem does this PR solve?

Issue Number: close #7486

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by windtalker — APPROVED** [review]
LGTM

File: dbms/src/Flash/Planner/PhysicalPlan.cpp [hunk 1]
```diff
@@ -76,6 +76,9 @@ void PhysicalPlan::build(const tipb::DAGRequest * dag_request)
 void PhysicalPlan::buildTableScan(const String & executor_id, const tipb::Executor * executor)
 {
     TiDBTableScan table_scan(executor, executor_id, dagContext());
+    if (!table_scan.getPushedDownFilters().empty() && unlikely(!context.getSettingsRef().dt_enable_read_thread))
+        throw Exception("Enable late materialization but disable read thread pool, please set the config `dt_enable_read_thread` of TiFlash to true,"
+                        "or disable late materialization by set tidb variable `tidb_opt_enable_late_materialization` to false.");
```
Comment (SeaRise) [inline dbms/src/Flash/Planner/PhysicalPlan.cpp:81]: Maybe this code needs to be added in `DAGQueryInterpreter` as well?

Comment (Lloyd-Pottiger) [reply dbms/src/Flash/Planner/PhysicalPlan.cpp:81]: done

**Review by SeaRise — APPROVED** [review]

**Pull request merged**